keras_cv/models/feature_extractor/clip/clip_image_model.py

```diff
@@ -0,0 +1,169 @@
+            shape=(
+                (
+                    (self.input_resolution // self.patch_size) ** 2 + 1,
+                    self.width,
+                )
+            ),
+            trainable=True,
+            name="patch_embed.positional_embedding",
+        )
+
+    def call(self, x):
+        batch_size, _, _, _ = ops.shape(x)
+        patch_embeddings = self.conv1(x)  # shape = [*, grid, grid, channel]
+
+        patch_embeddings = ops.reshape(
+            patch_embeddings, (batch_size, self.num_patches, -1)
+        )
+        class_embeds = ops.broadcast_to(
+            self.class_embedding, (batch_size, 1, self.width)
+        )
+        embeddings = ops.concatenate(
+            [class_embeds, patch_embeddings], axis=1
+        )  # shape = [*, grid ** 2 + 1, width]
+        positional_embedding = self.positional_embedding
+        embeddings = embeddings + positional_embedding
+        return embeddings
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "width": self.width,
+                "patch_size": self.patch_size,
+                "input_resolution": self.input_resolution,
+                "output_dim": self.output_dim,
+            }
+        )
+        return config
+
+
+class CLIPImageEncoder(keras.Model):
+    def __init__(
+        self,
+        input_resolution: int,
+        patch_size: int,
+        width: int,
+        num_layers: int,
+        heads: int,
+        output_dim: int,
+        **kwargs,
+    ):
+        super().__init__(
+            **kwargs,
+        )
+        self.input_resolution = input_resolution
+        self.width = width
+        self.patch_size = patch_size
+        self.output_dim = output_dim
+        self.heads = heads
+        self.num_layers = num_layers
+
+        self.embeddings = CLIPPatchingAndEmbedding(
+            width=self.width,
+            patch_size=self.patch_size,
+            input_resolution=self.input_resolution,
+            output_dim=self.output_dim,
+        )
+        self.pre_norm = keras.layers.LayerNormalization(
+            epsilon=1e-5, name="ln_1"
+        )
+        self.encoder = CLIPEncoder(
+            self.width,
+            self.num_layers,
+            self.heads,
+            name="clip_encoder",
+        )
+        self.post_norm = keras.layers.LayerNormalization(
+            epsilon=1e-5, name="ln_2"
+        )
+        self.image_projector = keras.layers.Dense(
+            output_dim, name="vision_projector", use_bias=False
+        )
+
+    def build(self, input_shape):
+        super().build(input_shape)
+        self.embeddings.build(input_shape)
+        self.pre_norm.build([None, None, self.width])
+        self.encoder.build(None)
+        self.post_norm.build([None, self.width])
+        self.image_projector.build([None, None, self.width])
+
+    def call(self, image):
+        embeddings = self.embeddings(image)
```

Comment:
could consider the `x = layer(x)` convention you did earlier, just to be consistent. no strong preference.

Reply:
done